compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1708,6 +1708,7 @@ private BType getEffectiveMappingType(BLangRecordLiteral recordLiteral, BType ap
         }
 
         recordType.fields = newFields;
+        recordType.flags = recordSymbol.flags;
```

Comment:
Should we use the `BRecordType` constructor that accepts flags also at L1670, instead? org.wso2.ballerinalang.compiler.semantics.model.types.BRecordType#BRecordType(org.wso2.ballerinalang.compiler.semantics.model.symbols.BTypeSymbol, int)